rfcs/rfc_001.md

```diff
@@ -0,0 +1,161 @@
+Ngeo/Gmf is added as a dependency and directly consummed as source code.
+There is no alternative frontend, no possibility to mix different versions or use other build tools.
+Notably, the code requires specific transformations, managed by webpack 4 plugins:
+- AngularJS dependency injection with ngInject;
+- SaaS files importation, unicity and ordering;
+- SVG files importation and build-time modification;
+- AngularJS partials importation.
+
+## Vision
+
+### Allow alternative frontends (GeoGirafe, MasterPortal, Uzuverse, other)
+
+- No change in the way geomapfish is deployed;
+- No change in the backend protocol;
+- integrate seemlesly with the existing backend and deployment mechanism;
+- Allow completly independant frontend (GeoGirafe, MasterPortal, Uzuverse, other);
+
+### Simplify Ngeo/Gmf build tools and make the code standard
+
+- Limit breaking changes for existing users;
+- Remove / simplify code;
+- Distribute the Ngeo/GMF library as pure ES6 + CSS package;
+
+### Allow to easily extend Ngeo/Gmf
+
+- Ability to use any build system, frontend framework, or package and in any version;
+- Expose some states on a case-by-case;
+- Document and demo this opt-in mechanism;
+
+
+## Technical proposition part 1: allow alternative frontends
+
+### Creation of a custom frontend
+
+There will be a `frontend` directory at the root of the Geomapfish project. This directory will contain the custom frontend project.
+
+In this folder the developer should add everything needed to build the frontend like a `package.json`, the custom components, the build tools, the interface html files, etc.
+
+This folder will contain a `Dockerfile`, in charge of building this custom frontend project.
+The content is up to the developer, the only constraint is to put the generated files in the `/etc/static-frontend` directory.
+
+Each interface html file must contain the header metadata `dynamicUrl`, pointing to the used `dynamic.json` like: `/dynamic.json` (used by ngeo).
+These interfaces will be served by the backend on `/` and `/theme/<name>`.
+
+The other files in the `frontend` directory will be served by the backend on the path `/static-frontend/<file>`.
+These paths should be correctly referenced in the html files.
+
+Note: above I consider the entry point as `/` but it should be correcty managed by the frontend Docker image if needed.
+
+Note: the i18n should be managed by the frontend project, the backend will only provide language files with localized string related to the backend (like the layer names).
+
+Default custom frontend: the demo will provide a working example, in addition, documentation will be added in the GeoMapFish documentation.
+
+### Integration of the custom frontend
+
+In `c2cgeoportal` we will add a new static route to serve the frontend files from `/etc/static-frontend` directory on the `/static-frontend` path.
+
+In the `c2cgeoportal` configuration we will add a new interface type `custom` (currently we have `ngeo` and `canvas`).
```

Comment:
Same remark as above. I'm not sure the backend should have any idea of the frontend used.
I would see the backend more as a service provider that can be consumed from any client.